lib/map/security_token/service.rb

```diff
@@ -2,21 +2,16 @@
 
 require 'map/security_token/configuration'
 require 'map/security_token/errors'
+require 'map/security_token/token_cache'
 
 module MAP
   module SecurityToken
     class Service < Common::Client::Base
       configuration Configuration
 
-      def token(application:, icn:)
+      def token(application:, icn:, cache: true)
         Rails.logger.info("#{config.logging_prefix} token request", { application:, icn: })
-        response = perform(:post,
-                           config.token_path,
-                           token_params(application, icn),
-                           { 'Content-Type' => 'application/x-www-form-urlencoded' })
-        sts_token = parse_response(response, application, icn)
-        Rails.logger.info("#{config.logging_prefix} token success", { application:, icn: })
-        sts_token
+        cache ? find_cached_token(application:, icn:) : request_token(application:, icn:)
```

Comment:
Although this works, I think this is a good use case for `Rails.cache`. It would greatly reduce config and overhead. You could basically just wrap this method in a `Rails.cache.fetch` with a force option.  
something like:

```ruby
def token(application:, icn:, cache: true)
  Rails.logger.info("#{config.logging_prefix} token request", { application:, icn: }) 

  Rails.cache.fetch("map_sts_token_#{application}_#{icn}", expires_in: 5.minutes, force: !cache) do
    response = perform(:post,
                       config.token_path,
                       token_params(application, icn),
                       { 'Content-Type' => 'application/x-www-form-urlencoded' })
    sts_token = parse_response(response, application, icn)
    Rails.logger.info("#{config.logging_prefix} token success", { application:, icn: })
    sts_token
  end
end
```

Reply:
I agree with using Rails cache, tbh it prob should be used in like 90% of our caching applications on `vets-api`, we're just so used to (including me) the custom caching that's implemented